code/core/src/main/java/com/adobe/marketing/mobile/rulesengine/RulesEngine.java

```diff
@@ -48,4 +49,10 @@ public void replaceRules(final List<T> newRules) {
             rules = newRules;
```

Comment:
This should store a copy of the newRules object instead of the reference as we modify the rules list in `addRules` function.